grails-core/src/main/groovy/org/grails/config/DotNotatedKeyParser.java

```diff
@@ -0,0 +1,46 @@
+package org.grails.config;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+public class DotNotatedKeyParser {
```

Comment:
Please make the class package private. No need to expose this API publicly. Also is a new class needed? A static method of NavigableMapConfig would probably be enough.

Reply:
Addressed in commit f2c6ba03a3c9c3a4a2a292bf3aa59a854201bb33